### ISSUE TITLE
Cannot access web UI on Heroku

### ISSUE THREAD

**Issue by NiccoMlt** [NiccoMlt/Protelis-Web#7]
```log
2019-12-18T10:59:16.246204+00:00 heroku[router]: at=error code=H20 desc="App boot timeout" method=GET path="/" host=protelis-web-develop.herokuapp.com request_id=277e4377-7d10-4895-b47f-6138def1c697 fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=https
2019-12-18T10:59:37.444643+00:00 heroku[web.1]: State changed from starting to crashed
2019-12-18T10:59:37.342362+00:00 heroku[web.1]: Error R10 (Boot timeout) -> Web process failed to bind to $PORT within 90 seconds of launch
2019-12-18T10:59:37.342362+00:00 heroku[web.1]: Stopping process with SIGKILL
2019-12-18T10:59:37.429771+00:00 heroku[web.1]: Process exited with status 137
2019-12-18T10:59:38.719214+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=protelis-web-develop.herokuapp.com request_id=cc0e5423-e740-48fb-bb32-51196d792e34 fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=https
2019-12-18T11:01:09.890847+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=protelis-web-develop.herokuapp.com request_id=5a7705e0-a75c-45f6-b3eb-be40c0239d3b fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=https
2019-12-18T11:01:10.264780+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=protelis-web-develop.herokuapp.com request_id=90298522-9dec-437e-bc97-da4e9a251d72 fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=https
2019-12-18T11:01:16.312098+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=protelis-web-develop.herokuapp.com request_id=fe6277e6-f781-4392-a8f9-f6cbbe760230 fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=https
2019-12-18T11:01:16.669072+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=protelis-web-develop.herokuapp.com request_id=bbfce6af-07fb-405c-a24e-f8c60c62d9f0 fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=https
2019-12-18T11:04:46.496455+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=protelis-web-develop.herokuapp.com request_id=fa4c58b9-0fc1-4c62-9707-56896bed714d fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=http
2019-12-18T11:04:47.178707+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=protelis-web-develop.herokuapp.com request_id=0bbc23cd-ea23-46e9-a3b4-1a2d891cd64d fwd="137.204.39.10" dyno= connect= service= status=503 bytes= protocol=http
```

**Comment by NiccoMlt** [comment]
Vert.x probably binds the wrong port

**Comment by NiccoMlt** [comment]
Fixed in 80328bcd0544ceb28276ebd64de431aeb480d8a9

**Comment by NiccoMlt** [comment]
Travis CI fails to deploy on Heroku:

> Preparing deploy
> 
> Cleaning up git repository with `git stash --all`. If you need build artifacts for deployment, set `deploy.skip_cleanup: true`. See https://docs.travis-ci.com/user/deployment#Uploading-Files-and-skip_cleanup.
> 
> No local changes to save
> 
> creating application archive
> 
> dpl.3
> 
> Deploying application
> 
> uploading application archive
> 
> triggering new deployment
> 
> -----> Gradle app detected
> 
> -----> Installing JDK 1.8... done
> 
> -----> Building Gradle app...
> 
> -----> executing ./gradlew stage
> 
>        Downloading https://services.gradle.org/distributions/gradle-6.1.1-all.zip
> 
>        ....................................................................................................................................
> 
>        To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/6.1.1/userguide/gradle_daemon.html.
> 
>        Daemon will be stopped at the end of the build stopping after processing
> 
>        > Task :buildSrc:compileKotlin
> 
>        > Task :buildSrc:compileJava NO-SOURCE
> 
>        > Task :buildSrc:compileGroovy NO-SOURCE
> 
>        > Task :buildSrc:pluginDescriptors
> 
>        > Task :buildSrc:processResources NO-SOURCE
> 
>        > Task :buildSrc:classes UP-TO-DATE
> 
>        > Task :buildSrc:inspectClassesForKotlinIC
> 
>        
> 
>        > Task :buildSrc:jar
> 
>        :jar: No valid plugin descriptors were found in META-INF/gradle-plugins
> 
>        
> 
>        > Task :buildSrc:assemble
> 
>        > Task :buildSrc:compileTestKotlin NO-SOURCE
> 
>        > Task :buildSrc:pluginUnderTestMetadata
> 
>        > Task :buildSrc:compileTestJava NO-SOURCE
> 
>        > Task :buildSrc:compileTestGroovy NO-SOURCE
> 
>        > Task :buildSrc:processTestResources NO-SOURCE
> 
>        > Task :buildSrc:testClasses UP-TO-DATE
> 
>        > Task :buildSrc:test NO-SOURCE
> 
>        > Task :buildSrc:validatePlugins
> 
>        > Task :buildSrc:check
> 
>        > Task :buildSrc:build
> 
>        > Task :clean UP-TO-DATE
> 
>        > Task :compileKotlin
> 
>        > Task :compileJava NO-SOURCE
> 
>        > Task :nodeSetup
> 
>        
> 
>        > Task :yarnSetup
> 
>        /tmp/build_0606b8cd065513ff27a61430c8658a6e/.gradle/yarn/yarn-v1.21.1/bin/yarn -> /tmp/build_0606b8cd065513ff27a61430c8658a6e/.gradle/yarn/yarn-v1.21.1/lib/node_modules/yarn/bin/yarn.js
> 
>        /tmp/build_0606b8cd065513ff27a61430c8658a6e/.gradle/yarn/yarn-v1.21.1/bin/yarnpkg -> /tmp/build_0606b8cd065513ff27a61430c8658a6e/.gradle/yarn/yarn-v1.21.1/lib/node_modules/yarn/bin/yarn.js
> 
>        + yarn@1.21.1
> 
>        updated 1 package in 0.592s
> 
>        
> 
>        > Task :yarn
> 
>        yarn install v1.21.1
> 
>        [1/4] Resolving packages...
> 
>        [2/4] Fetching packages...
> 
>        warning sha.js@2.4.11: Invalid bin entry for "sha.js" (in "sha.js").
> 
>        info fsevents@1.2.11: The platform "linux" is incompatible with this module.
> 
>        info "fsevents@1.2.11" is an optional dependency and failed compatibility check. Excluding it from installation.
> 
>        info fsevents@2.1.2: The platform "linux" is incompatible with this module.
> 
>        info "fsevents@2.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
> 
>        [3/4] Linking dependencies...
> 
>        [4/4] Building fresh packages...
> 
>        Done in 83.51s.
> 
>        
> 
>        > Task :buildFrontend
> 
>        yarn run v1.21.1
> 
>        $ react-scripts build
> 
>        Creating an optimized production build...
> 
>        Compiled successfully.
> 
>        
> 
>        File sizes after gzip:
> 
>        
> 
>          207.71 KB  build/static/js/2.8e45ed2e.chunk.js
> 
>          4.64 KB    build/static/js/main.d594132d.chunk.js
> 
>          770 B      build/static/js/runtime-main.1c2d59f4.js
> 
>        
> 
>        The project was built assuming it is hosted at https://protelis-web.herokuapp.com/.
> 
>        You can control this with the homepage field in your package.json.
> 
>        
> 
>        The build folder is ready to be deployed.
> 
>        You may serve it with a static server:
> 
>        
> 
>          yarn global add serve
> 
>          serve -s build
> 
>        
> 
>        Find out more about deployment here:
> 
>        
> 
>          bit.ly/CRA-deploy
> 
>        
> 
>        Done in 110.44s.
> 
>        
> 
>        > Task :copyToWebRoot
> 
>        > Task :processResources
> 
>        > Task :classes
> 
>        > Task :inspectClassesForKotlinIC
> 
>        > Task :jar
> 
>        > Task :startScripts
> 
>        > Task :distTar
> 
>        > Task :distZip
> 
>        > Task :shadowJar
> 
>        > Task :startShadowScripts
> 
>        > Task :shadowDistTar
> 
>        > Task :shadowDistZip
> 
>        > Task :assemble
> 
>        > Task :ktlintKotlinScriptCheck
> 
>        > Task :ktlintMainSourceSetCheck
> 
>        > Task :ktlintTestSourceSetCheck
> 
>        > Task :compileTestKotlin
> 
>        > Task :compileTestJava NO-SOURCE
> 
>        > Task :processTestResources
> 
>        > Task :testClasses
> 
>        
> 
>        > Task :test
> 
>        
> 
>        Gradle Test Executor 1 STANDARD_ERROR
> 
>            Feb 13, 2020 5:40:57 PM org.junit.platform.launcher.core.LauncherConfigurationParameters fromClasspathResource
> 
>            INFO: Loading JUnit Platform configuration parameters from classpath resource [file:/tmp/build_0606b8cd065513ff27a61430c8658a6e/build/resources/test/junit-platform.properties].
> 
>        
> 
>        > Task :test FAILED
> 
>        
> 
>        FAILURE: Build failed with an exception.
> 
>        
> 
>        * What went wrong:
> 
>        Execution failed for task ':test'.
> 
>        > There were failing tests. See the report at: file:///tmp/build_0606b8cd065513ff27a61430c8658a6e/build/reports/tests/test/index.html
> 
>        
> 
>        * Try:
> 
>        Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
>        
> 
>        * Get more help at https://help.gradle.org
> 
>        
> 
>        Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
> 
>        Use '--warning-mode all' to show the individual deprecation warnings.
> 
>        See https://docs.gradle.org/6.1.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
>        
> 
>        BUILD FAILED in 7m 40s
> 
>        23 actionable tasks: 22 executed, 1 up-to-date
> 
>  !     ERROR: Failed to run Gradle!
> 
>        We're sorry this build is failing. If you can't find the issue in application
> 
>        code, please submit a ticket so we can help: https://help.heroku.com
> 
>        You can also try reverting to the previous version of the buildpack by running:
> 
>        $ heroku buildpacks:set https://github.com/heroku/heroku-buildpack-gradle#previous-version
> 
>        
> 
>        Thanks,
> 
>        Heroku
> 
>  !     Push rejected, failed to compile Gradle app.
> 
>  !     Push failed
> 
> No stash entries found.
> 
> deploy failed
> 
> failed to deploy

**Comment by NiccoMlt** [comment]
Adding `system.properties` file with JDK11 requirement in commit 0ac9034d8de3b4467d723bd853df751e0098e275

**Comment by NiccoMlt** [comment]
It keeps failing tests: https://travis-ci.com/NiccoMlt/Protelis-Web/jobs/287066459

**Comment by NiccoMlt** [comment]
Now it fails because slug size is too big

**Comment by NiccoMlt** [comment]
I thought to 2 possible solutions:

1. resolve #10: taking out React resource project will slim down sizes a lot
2. try uploading only the shadow JAR, maybe built on Travis CI

**Comment by NiccoMlt** [comment]
Seems to be closed by 543b9ba51b934b9979197e5e66e36606c855566e
